includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -3430,11 +3430,6 @@ public function update_order_status() {
 				);
 			}
 
-			$payment_method_id = isset( $_POST['payment_method_id'] ) ? wc_clean( wp_unslash( $_POST['payment_method_id'] ) ) : '';
-			if ( 'null' === $payment_method_id ) {
-				$payment_method_id = '';
-			}
```

Comment:
`update_order_status()` is used to update 3DS orders after authentication. Shortcode checkout includes `payment_method_id` in the `update_order_status()` AJAX request for setup intents but blocks checkout doesn't. This payment method is only used in adding the token to the user. We need the token to determine the payment method type in order to correctly set the payment method title. This is all to say that we can safely remove this as it's now being set below, pulled from the intent.

Incidentally, this might solve #8611 but I haven't had a chance to verify that yet.